File: src/olcw78 (die yoong)/jsAlgorithmStudy/week01/03_두개_뽑아서_더하기.go
//package main
//
//import (
//	"fmt"
//	"slices"
//)
//
//func solution(numbers []int) []int {
//	unq := map[int]int{}
//	for i := 0; i < len(numbers); i++ {
//		for j := i; j < len(numbers); j++ {
//			if i == j {
//				continue
//			}
//
//			sum := numbers[i] + numbers[j]
//			unq[sum] = sum
//		}
//	}
//
//	arr := make([]int, 0, len(numbers))
//	for _, v := range unq {
//		arr = append(arr, v)
//	}
//	slices.SortFunc(arr, func(a, b int) int { return b - a })
//
//	return arr
//}
//
//func main() {
//	r1 := solution([]int{2, 1, 3, 4, 1})
//	fmt.Println(r1)
//
//	r2 := solution([]int{5, 0, 2, 7})
//	fmt.Println(r2)
//}

File: src/olcw78 (die yoong)/jsAlgorithmStudy/week01/03_두개_뽑아서_더하기.js
/**
 * @param numbers {number[]}
 */
function solution(numbers) {
    const result = new Set();
    for (let i = 0; i < numbers.length; i++) {
        for (let j = i; j < numbers.length; j++) {
            if (i === j) continue;

            const sum = numbers[i] + numbers[j];
            if (!result.has(sum)) {
                result.add(sum);
            }
        }
    }

    const arr = [...result];
    arr.sort((b, a) => b - a);
    return arr;
}

const r1 = solution([2, 1, 3, 4, 1]);
console.log(r1);

const r2 = solution([5, 0, 2, 7]);
console.log(r2);
